src/shared/pages/list/entityList/entityDefinitions/ukviEnquiryReason/__tests__/enquiryReasonView.spec.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { match, MemoryRouter } from 'react-router-dom';
+import { createBrowserHistory, History, Location } from 'history';
+import { act, wait, render, RenderResult } from '@testing-library/react';
+import EntityListView from '../../../entityListView';
+import * as ListService from '../../../../../../services/entityListService';
+import { State } from '../../../state';
+import * as useError from '../../../../../../hooks/useError';
+import enquiryReason from '../ukviEnquiryReason';
+
+let match: match<any>;
+let history: History<any>;
+let location: Location;
+let mockState: State;
+
+const CampaignsView = EntityListView(enquiryReason);
+
+jest.mock('../../../../../../services/entityListService', () => ({
+    __esModule: true,
+    getListItems: jest.fn().mockReturnValue(Promise.resolve({
+        data: [{
+            simpleName: 'testSimpleName1',
+            uuid: 'testId1',
+            title: 'testTitle1'
+        }, {
+            simpleName: 'testSimpleName2',
+            uuid: 'testId2',
+            title: 'testTitle2'
+        }]
+    }))
+}));
+
+const getListItemsSpy = jest.spyOn(ListService, 'getListItems');
+const useReducerSpy = jest.spyOn(React, 'useReducer');
+const useErrorSpy = jest.spyOn(useError, 'default');
+const setMessageSpy = jest.fn();
+const clearErrorsSpy = jest.fn();
+
+const renderComponent = () => render(
+    <MemoryRouter>
+        <CampaignsView history={history} location={location} match={match}></CampaignsView>
+    </MemoryRouter>
+);
+
+beforeEach(() => {
+    history = createBrowserHistory();
+    match = {
+        isExact: true,
+        params: {},
+        path: '',
+        url: ''
+    };
+
+    location = {
+        hash: '',
+        key: '',
+        pathname: '',
+        search: '',
+        state: {}
+    };
+    mockState = {
+        entitiesLoaded: true,
+        entities: [{
+            simpleName: 'testSimpleName1',
+            uuid: 'testId1',
+            title: 'testTitle1'
+        }, {
+            simpleName: 'testSimpleName2',
+            uuid: 'testId2',
+            title: 'testTitle2'
+        }]
+    };
+    useReducerSpy.mockImplementationOnce(() => [mockState, jest.fn()]);
+    useErrorSpy.mockImplementation(() => [{}, jest.fn(), clearErrorsSpy, setMessageSpy]);
+    clearErrorsSpy.mockReset();
+    setMessageSpy.mockReset();
+});
+
+describe('when the mpamCampaign EntityListView component is mounted', () => {
```

Comment:
Again can this be updated please to reference enq reason instead of mpamCampaign

Reply:
This has been corrected.